synapse/storage/databases/main/registration.py

```diff
@@ -166,6 +166,42 @@ async def get_user_by_id(self, user_id: str) -> Optional[Dict[str, Any]]:
             desc="get_user_by_id",
         )
 
+    @cached()
+    async def get_userinfo_by_id(self, user_id: str) -> Optional[UserInfo]:
```

Comment:
please can you give this a docstring?

Also, please can you stick a comment or docstring on get_user_by_id to say it's deprecated and replaced by get_userinfo_by_id?

Reply:
> Probably best deprecate _store.get_user_by_id in a follow-up pr, I'm happy to do that?

Oh I just read this. I'd roll it into this PR, since it's simple enough.

Unless you were talking about actually replacing uses of `get_user_by_id` with `get_userinfo_by_id`? Which would be amazing if you've got time, and yes a separate PR would be best.